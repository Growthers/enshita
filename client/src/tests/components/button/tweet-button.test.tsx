import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import TweetButton from "~/components/button/tweet-button";

describe("(component) TweetButton", () => {
  test("Snap Shot - Only Text", () => {
    const { container } = render(<TweetButton text="Test Tweet Content" />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - Contain URL", () => {
    const { container } = render(
      <TweetButton text="Test Tweet Content" url="https://growthers.dev" />,
    );
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - Contain Hashtags and URL", () => {
    const { container } = render(
      <TweetButton
        text="Test Tweet Content"
        url="https://growthers.dev"
        hashtags={["共同開発鯖", "#共同開発鯖LT"]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
