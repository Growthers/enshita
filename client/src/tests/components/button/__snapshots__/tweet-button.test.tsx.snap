// Vitest Snapshot v1

exports[`(component) TweetButton > Snap Shot - Contain Hashtags 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 _enshita-event-tweet-box_1hg6k_5"
    href="https://twitter.com/intent/tweet?text=Test%20Tweet%20Content&url=&hashtags=共同開発鯖"
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <div
      class="_enshita-button-icon-box_8w723_1 _enshita-event-tweet-icon_1hg6k_1"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_8w723_36 _enshita-event-tweet-text_1hg6k_13"
    >
      Tweet
    </p>
  </a>
</div>
`;

exports[`(component) TweetButton > Snap Shot - Contain Hashtags and URL 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 _enshita-event-tweet-box_1hg6k_5"
    href="https://twitter.com/intent/tweet?text=Test%20Tweet%20Content&url=https://growthers.dev&hashtags=共同開発鯖,共同開発鯖LT"
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <div
      class="_enshita-button-icon-box_8w723_1 _enshita-event-tweet-icon_1hg6k_1"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_8w723_36 _enshita-event-tweet-text_1hg6k_13"
    >
      Tweet
    </p>
  </a>
</div>
`;

exports[`(component) TweetButton > Snap Shot - Contain URL 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 _enshita-event-tweet-box_1hg6k_5"
    href="https://twitter.com/intent/tweet?text=Test%20Tweet%20Content&url=https://growthers.dev&hashtags="
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <div
      class="_enshita-button-icon-box_8w723_1 _enshita-event-tweet-icon_1hg6k_1"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_8w723_36 _enshita-event-tweet-text_1hg6k_13"
    >
      Tweet
    </p>
  </a>
</div>
`;

exports[`(component) TweetButton > Snap Shot - Only Text 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 _enshita-event-tweet-box_1hg6k_5"
    href="https://twitter.com/intent/tweet?text=Test%20Tweet%20Content&url=&hashtags="
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <div
      class="_enshita-button-icon-box_8w723_1 _enshita-event-tweet-icon_1hg6k_1"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_8w723_36 _enshita-event-tweet-text_1hg6k_13"
    >
      Tweet
    </p>
  </a>
</div>
`;
